practice_app/pages/api/getSimilarProducts.js

```diff
@@ -0,0 +1,102 @@
+const getSimilarProducts = (req, res) =>
+  new Promise(async (resolve) => {
+    // Imports the Google Cloud client library
+    const vision = require("@google-cloud/vision");
+    // Id of the getflix in Google Cloud Storage
+    const projectId = "brilliant-era-276800";
+    // Location of the getflix in Google Cloud Storage
+    const location = "europe-west1";
+    // Imports file system module to read images
+    const fs = require("fs");
+    // Creates a product client
+    const productSearchClient = new vision.ProductSearchClient();
+    // Creates an image client
+    const imageAnnotatorClient = new vision.ImageAnnotatorClient();
+    // Gets url of the current page
+    var requestURL = req.url;
+    // Parses url into query parameters
+    var initial = requestURL.indexOf("?");
+    initial = initial + 1;
+    requestURL = requestURL.substring(initial);
+    requestURL = requestURL.split("&");
+    var queryParams = {};
+    requestURL.forEach((pair) => {
+      var keyValue = pair.split("=");
+      if (keyValue[1].includes("%3D")) {
+        keyValue[1] = keyValue[1].split("%3D").join("=");
+      }
+      if (keyValue[1].includes("%3A")) {
+        keyValue[1] = keyValue[1].split("%3A").join(":");
+      }
+      if (keyValue[1].includes("%2F")) {
+        keyValue[1] = keyValue[1].split("%2F").join("/");
+      }
+      if (keyValue[1].includes("%5C")) {
+        keyValue[1] = keyValue[1].split("%5C").join("\\");
+      }
+      queryParams[keyValue[0]] = keyValue[1];
+    });
```

Comment:
Please do not reinvent the wheel by trying to parse url query params yourself. This is brittle code. There are already libraries out there that do this. Try https://www.npmjs.com/package/query-string